subsys/bluetooth/mesh/light_hsl_srv.c

```diff
@@ -559,7 +559,11 @@ static int bt_mesh_light_hsl_srv_start(struct bt_mesh_model *model)
 			/* The lightness server updated its "last" value in its
 			 * start function, which has already completed:
 			 */
-			.lightness = srv->lightness->last,
+			.lightness =
+				(atomic_test_bit(&srv->lightness->flags, LIGHTNESS_SRV_FLAG_IS_ON) ?
+					       srv->lightness->last :
+					       0),
```

Comment:
Good catch, makes sense

Reply:
Thanks for catching this 👍 
Just minor suggestion. Since the lightness server has set the light value through the callback, I think HSL server can call `lightness->handlers->light_get()` to retrieve the light value, instead of touching the last value and the lightness's flags. E.g. XYL server does this through `xyl_get()`.